tools/accuracy_checker/accuracy_checker/evaluators/quantization_model_evaluator.py

```diff
@@ -116,19 +115,34 @@ def _create_subset(subset, num_images):
             if ready_irs:
                 wait_time = 0.01
                 while ready_irs:
-                    batch_id, batch_annotation, batch_identifiers, batch_meta, batch_predictions, ir = ready_irs.pop(0)
+                    ready_data = ready_irs.pop(0)
+                    (
+                        batch_id,
+                        batch_input_ids,
+                        batch_annotation,
+                        batch_identifiers,
+                        batch_meta,
+                        batch_raw_predictions,
+                        ir
+                    ) = ready_data
                     batch_predictions = _process_ready_predictions(
-                        batch_predictions, batch_identifiers, batch_meta, self.adapter, kwargs.get('output_callback')
+                        batch_raw_predictions, batch_identifiers, batch_meta, self.adapter
                     )
                     free_irs.append(ir)
                     annotations, predictions = self.postprocessor.process_batch(batch_annotation, batch_predictions)
 
+                    metrics_result = None
                     if self.metric_executor:
-                        self.metric_executor.update_metrics_on_batch(annotations, predictions)
+                        metrics_result = self.metric_executor.update_metrics_on_batch(
+                            batch_input_ids, annotations, predictions
+                        )
                         if self.metric_executor.need_store_predictions:
                             self._annotations.extend(annotations)
                             self._predictions.extend(predictions)
 
+                    if output_callback:
+                        output_callback(batch_raw_predictions, metrics_result=metrics_result)
```

Comment:
2 @eaidova:  Ekaterina, could you also pass to `output_callback` a list that will allow to match for each element in the batch
* "element identifier" -- i.e. the string that identifies the element in the dataset (most probably they are in `batch_identifiers`, aren't they?)
* "element index" -- i.e. the number of the element in the dataset

Reply:
done,
added `element_identifiers` argument for `batch_identifiers` and `dataset_indices` for dataset elements numeration.